### PR TITLE
make mytime less busy

### LINKED ISSUE #1624: Make Mytime less busy
**Is your feature request related to a problem? Please describe.**
It's very nice when a screen you use frequently is well-designed - and not too busy.

**Describe the solution you'd like**
Don't show hours by default.  
Instead only show the hours, and the options once you click on the Task.  
If you click again you go to the Goal as usual - unless you use any of the Task actions.

**Describe alternatives you've considered**
None. It's an experiment.

**Additional context**
image

### PR DESCRIPTION
**Pull request by vinaybadgujar102**
resolves #1624 
also separated UI elements such as GoalTiming and TaskOption from MyTimeline into individual components for improved maintainability and readability.

### CODE REVIEW

File: src/components/MyTimeComponents/MyTimelineElements.tsx
Comment (Tushar-4781) [inline src/components/MyTimeComponents/MyTimelineElements.tsx:1]: @vinaybadgujar102 please split these components in separate files. file cannot have more than 1 main component

**Review by Tushar-4781 — COMMENTED** [review]
@tijlleenders please confirm the UI

File: src/components/MyTimeComponents/MyTimeline/GoalTiming.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import React from "react";
+
+interface GoalTimingProps {
+  startTime: string | null;
+  endTime: string | null;
+}
+
+export const GoalTiming: React.FC<GoalTimingProps> = ({ startTime, endTime }) => {
+  const renderTime = (time: string | null) =>
```
Comment (Tushar-4781) [inline src/components/MyTimeComponents/MyTimeline/GoalTiming.tsx:9]: make it simple. accept time only. don't call the function if your value is null

File: src/components/MyTimeComponents/MyTimeline/GoalTiming.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    time ? (
+      <>
+        <span>{parseInt(time, 10)}</span>
+        <sup>00</sup>
+      </>
+    ) : null;
+
+  return (
+    <div className="MTL-goalTiming">
+      {renderTime(startTime)}
```
Comment (Tushar-4781) [inline src/components/MyTimeComponents/MyTimeline/GoalTiming.tsx:19]: do conditional rendering here

**Review by Tushar-4781 — CHANGES_REQUESTED** [review]
Please do the following changes & resolve merge conflicts and we are good to go

**Comment by tijlleenders** [comment]
@vinaybadgujar102 Please take PR out of draft mode and ask Tushar to review when you're done.

**Review by Tushar-4781 — APPROVED** [review]

**Pull request merged**